src/exchange/exchange.provider.ts

```diff
@@ -113,4 +116,10 @@ export class ExchangeProvider {
   private hasAllReqCreds(auth: ExchangeAuth, reqKeys: string[]) {
     return reqKeys.filter((key) => Boolean(auth[key])).length === reqKeys.length
   }
+
+  private error(error: string) {
+    logger.error(error)
+    wsServer.broadcast('error', { error })
```

Comment:
Here you will definitely need some kind of error class, that you know, that the error is sent by ExchangeProdiver.